mhs/common/mhs_common/state/tests/test_work_description.py

```diff
@@ -236,3 +289,49 @@ def test_create_wd_null_parameters(self):
                     outbound_status=wd.MessageStatus.OUTBOUND_MESSAGE_RECEIVED,
                     workflow=None
                 )
+        with self.subTest('Null statuses'):
+            with self.assertRaises(ValueError):
+                wd.create_new_work_description(
+                    persistence,
+                    key='aaa',
+                    outbound_status=None,
+                    inbound_status=None,
+                    workflow=workflow.SYNC_ASYNC
+                )
+
+        with self.subTest('Single null status should not raise error'):
+            wd.create_new_work_description(
+                persistence,
+                key='aaa',
+                outbound_status=wd.MessageStatus.INBOUND_RESPONSE_FAILED,
+                inbound_status=None,
+                workflow=workflow.SYNC_ASYNC
+            )
+
+class TestWorkDescriptionStatusUpdateRetry(unittest.TestCase):
+
+    @test_utilities.async_test
+    async def test_update_success(self):
+        wdo = MagicMock()
+        wdo.update.return_value = test_utilities.awaitable(True)
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(True)
+        await wd.update_status_with_retries(wdo, wdo.set_outbound_status,
+                                            wd.MessageStatus.OUTBOUND_MESSAGE_ACKD,
+                                            20)
+        self.assertEqual(wdo.update.call_count, 1)
+        self.assertEqual(wdo.set_outbound_status.call_count, 1)
+
+    @test_utilities.async_test
+    async def test_update_correct_number_retries(self):
+        async def throws_out_of_date_exception(param):
+            raise wd.OutOfDateVersionError
+
+        wdo = MagicMock()
+        wdo.update.return_value = test_utilities.awaitable(True)
+        wdo.set_outbound_status.side_effect = throws_out_of_date_exception
```

Comment:
Not something you have to change now, but I think you could have just specified `wd.OutOfDateVersionError` as the side-effect here, and it would be raised.

Reply:
Thought this had to be set so that the call something that can be awaited but I've just tested it and it seems it doesn't matter  so changed it to your suggestion